src/main/java/io/cryostat/rules/PeriodicArchiver.java

```diff
@@ -105,50 +98,30 @@ public void run() {
-        // FIXME using an HTTP request to localhost here works well enough, but is needlessly
-        // complex. The API handler targeted here should be refactored to extract the logic that
-        // creates the recording from the logic that simply figures out the recording parameters
-        // from the POST form, path param, and headers. Then the handler should consume the API
-        // exposed by this refactored chunk, and this refactored chunk can also be consumed here
-        // rather than firing HTTP requests to ourselves
+    void performArchival() throws InterruptedException, ExecutionException, Exception {
 
+        String recordingName = rule.getRecordingName();
+        ConnectionDescriptor connectionDescriptor =
+                new ConnectionDescriptor(serviceRef, credentialsManager.getCredentials(serviceRef));
         String path =
                 URI.create(
                                 String.format(
                                         "/api/v1/targets/%s/recordings/%s",
                                         URLEncodedUtils.formatSegments(
                                                 serviceRef.getServiceUri().toString()),
-                                        URLEncodedUtils.formatSegments(rule.getRecordingName())))
+                                        URLEncodedUtils.formatSegments()))
                         .normalize()
                         .toString();
+
         this.logger.trace("PATCH \"save\" {}", path);
 
-        CompletableFuture<String> future = new CompletableFuture<>();
-        this.webClient
-                .patch(path)
-                .putHeaders(headersFactory.apply(credentialsManager.getCredentials(serviceRef)))
-                .sendBuffer(
-                        Buffer.buffer("save"),
-                        ar -> {
-                            if (ar.failed()) {
-                                this.logger.error(
-                                        new IOException("Periodic archival failed", ar.cause()));
-                                future.completeExceptionally(ar.cause());
-                                return;
-                            }
-                            HttpResponse<Buffer> resp = ar.result();
-                            if (!HttpStatusCodeIdentifier.isSuccessCode(resp.statusCode())) {
-                                this.logger.error(resp.bodyAsString());
-                                future.completeExceptionally(new IOException(resp.bodyAsString()));
-                                return;
-                            }
-                            future.complete(resp.bodyAsString());
-                        });
-        this.previousRecordings.add(future.get());
+        String saveName = recordingHelper.saveRecording(connectionDescriptor, recordingName);
+        this.previousRecordings.add(saveName);
     }
 
     Future<Boolean> pruneArchive(String recordingName) {
```

Comment:
This should probably be public (anything not `private` in here should most likely be `public`).

Also, everything `public` in here is some kind of blocking operation - either filesystem access or network access, or both. So, I would say either every method should return a `Future` or none should. I think I would prefer them all to return `Future`s so that it's made clear that these are all potentially long operations and whoever is calling them should be doing so asynchronously/on some worker thread.